column-grouped/parseData.js

```diff
@@ -9,23 +9,42 @@ import * as d3 from 'd3';
  * @param  {String} url Path to CSV file
  * @return {Object}     Object containing series names, value extent and raw data object
  */
-export function fromCSV(url) {
+export function fromCSV(url, options) {
     return new Promise((resolve, reject) => {
         d3.csv(url, (error, data) => {
             if (error) reject(error);
             else {
-
+                const { sort, sortOn } = options;
                 const seriesNames = getSeriesNames(data.columns);
 
+                const groupNames = data.map(d => d.name).filter(d => d); // create an array of the group names
+                // Use the seriesNames array to calculate the minimum and max values in the dataset
+
                 // Use the seriesNames array to calculate the minimum and max values in the dataset
                 const valueExtent = extentMulti(data, seriesNames);
 
-                const columnNames = data.map(d => d.name); // create an array of the column names
+                // Buid the dataset for plotting
+                const plotData = data.map(d => ({
+                    name: d.name,
+                    groups: getGroups(seriesNames, d),
+                }));
+
+                if (sort === 'descending') {
+                    plotData.sort((a, b) =>
+                    // console.log("sortON=",sortOn)
+                    // console.log("SortOn",a.groups[sortOn],a.groups[sortOn].value,b.groups[sortOn],b.groups[sortOn].value)
+                        b.groups[sortOn].value - a.groups[sortOn].value);// Sorts biggest rects to the left
```

Comment:
Probably worth removing the extraneous commented code above.

Reply:
done